### PR TITLE
Prototype use of DataChannelRereader in EV3GyroSensor

### PR DESCRIPTION
**Pull request by dwalend**
Prototype of using DataChannelRereader in EV3Gyro

### CODE REVIEW

**Comment by jabrena** [comment]
Reviewing code

File: src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java [hunk 1]
```diff
@@ -82,9 +85,9 @@ public SampleProvider getRateMode() {
      * @return A sampleProvider
      *     See {@link lejos.robotics.SampleProvider leJOS conventions for SampleProviders}
      */
-    public SampleProvider getAngleMode() {
+    public EV3GyroAngleMode getAngleMode() {
```
Comment (jabrena) [inline src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java:88]: Every mode returns a `SampleProvider`

File: src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java [hunk 2]
```diff
@@ -118,4 +121,53 @@ public void reset() {
         switchMode(MODE_RATE_ANGLE, SWITCH_DELAY);
     }
 
+    public static class EV3GyroAngleMode implements SensorMode {
```
Comment (jabrena) [inline src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java:124]: That mode is not necessary. I understand that you added the method getAngle() but it breaks the contract.
It is necessary to upgrade with DataChannelRereader: 
https://github.com/ev3dev-lang-java/ev3dev-lang-java/blob/master/src/main/java/ev3dev/sensors/GenericMode.java

File: src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java [hunk 3]
```diff
@@ -82,9 +85,9 @@ public SampleProvider getRateMode() {
      * @return A sampleProvider
      *     See {@link lejos.robotics.SampleProvider leJOS conventions for SampleProviders}
      */
-    public SampleProvider getAngleMode() {
+    public EV3GyroAngleMode getAngleMode() {
         switchMode(MODE_ANGLE, SWITCH_DELAY);
-        return getMode(1);
+        return (EV3GyroAngleMode)getMode(1);
```
Comment (jabrena) [inline src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java:90]: It is not necessary to change that line

**Review by jabrena — COMMENTED** [review]
Waiting for changes

File: src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java (hunk 1 above)
Comment (dwalend) [reply src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java:88]: EV3GyroAngleMode is a SampleProvider.

File: src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java (hunk 2 above)
Comment (dwalend) [reply src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java:124]: In what way does it break the contract?

Touching GenericMode is a much broader change. I'll put in a pull request to show that.

**Comment by jabrena** [comment]
Old proposal

**Pull request closed without merging**